spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/s3/S3AutoConfiguration.java

```diff
@@ -78,8 +82,14 @@ S3ClientBuilder s3ClientBuilder(AwsClientBuilderConfigurer awsClientBuilderConfi
 	@ConditionalOnMissingBean(S3Operations.class)
 	@ConditionalOnBean(S3ObjectConverter.class)
 	S3Template s3Template(S3Client s3Client, S3OutputStreamProvider s3OutputStreamProvider,
-			S3ObjectConverter s3ObjectConverter) {
-		return new S3Template(s3Client, s3OutputStreamProvider, s3ObjectConverter);
+			S3ObjectConverter s3ObjectConverter, S3Presigner s3Presigner) {
+		return new S3Template(s3Client, s3OutputStreamProvider, s3ObjectConverter, s3Presigner);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	S3Presigner s3Presigner(S3Properties properties) {
+		return S3Presigner.builder().serviceConfiguration(properties.toS3Configuration()).build();
```

Comment:
we also must set here credentials and region provider, endpoint override, fips enabled etc. Same things as we do to all the clients in `AwsClientBuilderConfigurer#configure`.

Reply:
Let me know if I did this right, I was a bit unsure.